services/net/clip/ClipAction.cs

```diff
@@ -86,18 +87,16 @@ public ClipAction(IApiService api, IOptions<ClipOptions> options, ILogger<ClipAc
                         // more than an 2 minutes old. Assumption is that it is stuck.
                         reference = await this.UpdateContentReferenceAsync(reference, WorkflowStatus.InProgress);
                     }
-                    else reference = null;
+                    else continue;
 
-                    if (reference != null)
-                    {
-                        // TODO: Waiting for each clip to complete isn't ideal.  It needs to handle multiple processes.
-                        await RunProcessAsync(process, cancellationToken);
-                        await this.ContentReceivedAsync(manager, reference, CreateSourceContent(process, manager.Ingest, schedule, reference));
+                    // TODO: Waiting for each clip to complete isn't ideal.  It needs to handle multiple processes.
+                    await RunProcessAsync(process, cancellationToken);
 
-                        // Delete any schedule marked with RunOnlyOnce
-                        if (schedule.RunOnlyOnce)
-                            await this.DeleteSchedule(new IngestScheduleModel(manager.Ingest.Id, schedule));
-                    }
+                    await ContentReceivedAsync(manager, reference, CreateSourceContent(process, manager.Ingest, schedule, reference));
```

Comment:
I think I see what you mentioned in another comment.  If `RunProcessAsync` takes a while it means the `reference` is old.  You will have to make a request for the `reference` after the `RunProcessAsync` and then handle null.

Reply:
Updated, thanks